### ISSUE TITLE
Remove device does not clean up the agent

### ISSUE THREAD

**Issue by jwillmer** [shellhub-io/shellhub#213]
I dont know if this is a bug or a missing feature but after deleting a device from the web ui I noticed that the docker agent is still running on my client. I think the agent should remove itself when the device was removed.

**Comment by otavio** [comment]
@jwillmer I am wondering if this should be:

1. enforced
1. optional

and if this is optional, if this should be done by default or not.

**Comment by microhobby** [comment]
I had the same experience here.  I was expecting it to be removed from the board when using the remove option from the UI.  In the UI as the board is still running the container after a refresh the item returns to the list of Devices.

**Comment by otavio** [comment]
Good and about the comment I did above?

Enforced or optional?

**Comment by microhobby** [comment]
I think it may be optional, but the remove should be the default option.  If the user wants to keep it on the device he can choose to do so. But this is the opinion based on what I expected first, and thinking on the user experience with other systems this make sense to me, it is what my mind expects to happen.

**Comment by jwillmer** [comment]
Yes, I also think if I choose to remove the device it should cleanup the docker container. Would there be a use case for keeping the docker container?

**Comment by otavio** [comment]
Imagine you remove the wrong device and it is a device that is not easy to have physical access to it. It'll be a problem.

We also have use cases where the agent is not running inside a Docker container, for example running inside an embedded device that was built using the Yocto Project. In the case, the agent cannot be removed (possibly it runs inside a read-only rootfs).

I'd imagine we could "block" a device from adding itself (which could reduce the keepalive interval), and for blocked devices and offer an option to uninstall the agent if, and only if, it is running on Docker.

**Comment by jwillmer** [comment]
Maybe a two step approach? 
1) deactivate device / invalidate/block client access
2) option to delete device (if in state deactivated & can be deleted) - otherwise gray delete option with note that it can't be deleted because X

**Comment by otavio** [comment]
We did experiment doing it but there are a number of cases it can go wrong. Also, especially embedded devices, it can be stored on read-only storage and we'd need to have different behavior for Docker vs native packages and embedded.

We believe if the user wishes to remove the agent, it can be done using the tunnel itself. Also, we improved the device registration workflow so now the devices return to pending state now.
